From 32135702ea15c977485ca66cfedd1410896c895c Mon Sep 17 00:00:00 2001
From: Fangrui Song <maskray@google.com>
Date: Fri, 25 Mar 2022 17:18:30 -0700
Subject: [PATCH 229/249] apparmor: Fix -Wfortify-source for Clang

```
criu/apparmor.c:679:26: error: 'fscanf' may overflow; destination buffer in argument 3 has size 48, but the corresponding specifier may require size 49 [-Werror,-Wfortify-source]
        ret = fscanf(f, "%48s", contents);
```
The buffer size should be at least one larger than the fscanf maximum
field width.

Fixes: 8d992a680ef3 ("lsm: support checkpoint/restore of stacked apparmor profiles")
Signed-off-by: Fangrui Song <maskray@google.com>
---
 criu/apparmor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/criu/apparmor.c b/criu/apparmor.c
index f9ad79619..cc9e7ac2a 100644
--- a/criu/apparmor.c
+++ b/criu/apparmor.c
@@ -668,7 +668,7 @@ int dump_aa_namespaces(void)
 
 bool check_aa_ns_dumping(void)
 {
-	char contents[48];
+	char contents[49];
 	int major, minor, ret;
 	FILE *f;
 
-- 
2.35.1

